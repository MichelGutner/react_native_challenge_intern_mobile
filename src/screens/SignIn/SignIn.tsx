import React, { useState } from 'react';
import { StatusBar, Text, Alert, View, TouchableOpacity } from 'react-native';
import { Themes } from '../../../themes/themes';
import SignInputs from '../../components/Inputs/SignInputs';
import SignButton from '../../components/SignButton/SignButton';
import {
  Container,
  ImageLogoSignIn,
  ImageProfile,
  InputAreaView,
  MessageRegisterView,
  RegisterTextView,
  SignUpTouchable,
  TitleSignUpTouchable,
} from './styles';

// import { handleRequeriRegisterButton, handleSignIn } from './bussiness';
import { doLogin } from '../../services/loginAPI';
import { useNavigation } from '@react-navigation/native';

const messageRegister = 'Não possui um acesso?';
const messageRegisterButton = 'Cadastre-se aqui';

const SignIn = () => {
  const navigation = useNavigation();
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');

  const handleLogin = async () => {
    const result = await doLogin({ email: email, password: password });
    if (result === 'FORBIDDEN') {
      Alert.alert('resultado', result);
    } else {
      Alert.alert('Entrou');
    }
    // const listErrors = validateLogin(email, password);
    // console.log(listErrors);
    // if (listErrors.length >= 0) {
    //   console.log('e', 'erro');
    // }
  };

  const validateLogin = (email: string, password: string) => {
    const listErrors = [];
    // bloco email
    if (email === null || email === '') {
      for (let i = 0; i < email.length; i++) {
        const character = email.charAt(i);
        if (character === '@') {
          listErrors.push('O e-mail deve possuir arroba(@)');
        }

        if (character === '@') {
          listErrors.push('O e-mail deve possuir arroba(@)');
        }
      }

      listErrors.push('Validação de email');
    }

    // bloco password
    if (password === null || password === '') {
      let letterUpper = false;
      let letterLower = false;
      let haveNumber = false;
      let haveSpecialCharacter = false;

      // Mínimo de 8 caracteres.
      if (password.length < 8) {
        listErrors.push('A senha precisa de pelo menos 8 caracteres');
      }

      // Pelo menos 1 caractere maiúsculo.
      // Pelo menos 1 caractere minúsculo.
      // Pelo menos 1 dígito.
      for (let i = 0; i < password.length; i++) {
        // UpperCase
        const character = password.charAt(i);
        if (
          character === character.toUpperCase() &&
          character !== character.toLowerCase()
        )
          letterUpper = true;

        // LowerCase
        if (
          character === character.toLowerCase() &&
          character !== character.toUpperCase()
        )
          letterLower = true;

        // IsNumeric
        if (!isNaN(parseFloat(character)) && isFinite(parseInt(character))) {
          haveNumber = true;
        }

        if (!/^[a-zA-Z0-9]+$/.test(character)) {
          haveSpecialCharacter = true;
        }
      }

      if (!letterUpper)
        listErrors.push('A senha precisa de pelo menos 1 caracter maiusculo');

      if (!letterLower)
        listErrors.push('A senha precisa de pelo menos 1 caracter minusculo');

      if (!haveNumber) {
        listErrors.push('A senha precisa de pelo menos 1 digito');
      }

      if (haveSpecialCharacter) {
        listErrors.push('A senha precisa de pelo menos 1 caracter especial');
      }
    }
    return listErrors;
  };

  return (
    <Container>
      <StatusBar barStyle="light-content" />
      <ImageLogoSignIn
        source={require('../../components/SanarLogo/Logo-negativo.png')}
      />
      <ImageProfile
        source={require('../../components/SanarLogo/profile.png')}
      />
      <InputAreaView>
        <SignInputs
          placeholderTextColor={Themes.colors.whiteOpacity}
          placeholder="Digite seu e-mail"
          value={email}
          onChangeText={(text: string) => setEmail(text)}
        />
        <SignInputs
          placeholderTextColor={Themes.colors.whiteOpacity}
          placeholder="Digite sua senha de acesso"
          value={password}
          onChangeText={(text: string) => setPassword(text)}
          secureTextEntry={true}
        />
      </InputAreaView>
      <SignButton onPress={handleLogin} title="Entrar" />
      <RegisterTextView>
        <MessageRegisterView>
          <Text>{messageRegister}</Text>
        </MessageRegisterView>
        <SignUpTouchable onPress={() => navigation.navigate('SignUp')}>
          <TitleSignUpTouchable>{messageRegisterButton}</TitleSignUpTouchable>
        </SignUpTouchable>
      </RegisterTextView>
    </Container>
  );
};

export default SignIn;
