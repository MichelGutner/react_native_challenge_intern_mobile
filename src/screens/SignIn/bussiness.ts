import { doLogin } from '../../services/loginAPI';

type Props = {
  email: string;
  password: string;
  onSuccess: () => void;
  onError: () => void;
};

export const doLoginRequest = async ({
  email,
  password,
  onSuccess,
  onError,
}: Props) => {
  const result = await doLogin({
    email,
    password,
    onSuccess,
    onError,
  });
  if (result === 'FORBIDDEN') {
    onError();
  } else {
    onSuccess();
  }
};
