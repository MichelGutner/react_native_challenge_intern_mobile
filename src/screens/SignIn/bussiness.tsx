import { useNavigation } from '@react-navigation/native';

export const handleRequeriRegisterButton = () => {
  const navigation = useNavigation();
  navigation.navigate('SignIn');
};
