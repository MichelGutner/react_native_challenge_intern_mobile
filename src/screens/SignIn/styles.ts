import styled from 'styled-components/native';
import { Themes } from './../../../themes/themes';

export const Container = styled.SafeAreaView`
  flex: 1;
  background-color: ${Themes.colors.backgroundSanar};
`;

export const ImageLogoSignIn = styled.Image`
  width: 152px;
  height: 45px;
  left: 104px;
  margin-top: 96px;
`;
export const ImageProfile = styled.Image`
  width: 94px;
  height: 94px;
  left: 133px;
  top: 44px;
`;

export const InputAreaView = styled.View`
  left: 24px;
  top: 55px;
`;

export const SignUpTouchable = styled.TouchableOpacity`
  margin-left: 5px;
`;

export const TitleSignUpTouchable = styled.Text`
  font-family: Red Hat Display;
  font-style: normal;
  font-weight: normal;
  font-size: 14px;

  text-align: center;

  color: ${Themes.colors.white};
`;

export const RegisterTextView = styled.View`
  flex-direction: row;
  width: 100%;
  height: 24px;
  top: 110px;
  justify-content: center;
`;

export const MessageRegisterView = styled.Text`
  font-family: Red Hat Display;
  font-style: normal;
  font-weight: normal;
  font-size: 14px;

  text-align: center;

  color: ${Themes.colors.whiteOpacity};
`;
