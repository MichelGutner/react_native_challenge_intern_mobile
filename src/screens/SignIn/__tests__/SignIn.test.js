import { NavigationContainer } from '@react-navigation/native';
import React from 'react';
import renderer from 'react-test-renderer';
import SignIn from '../SignIn';

test('renders correctly', () => {
  const tree = renderer
    .create(
      <NavigationContainer>
        <SignIn />
      </NavigationContainer>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
