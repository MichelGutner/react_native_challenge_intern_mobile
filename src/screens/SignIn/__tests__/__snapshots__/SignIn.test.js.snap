// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "backgroundColor": "#0050F0",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/components/SanarLogo/Logo-negativo.png",
      }
    }
    style={
      Array [
        Object {
          "height": 45,
          "left": 104,
          "marginTop": 96,
          "width": 152,
        },
      ]
    }
  />
  <Image
    source={
      Object {
        "testUri": "../../../src/components/SanarLogo/profile.png",
      }
    }
    style={
      Array [
        Object {
          "height": 94,
          "left": 133,
          "top": 44,
          "width": 94,
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "left": 24,
          "top": 55,
        },
      ]
    }
  >
    <TextInput
      onChangeText={[Function]}
      placeholder="Digite seu e-mail"
      placeholderTextColor="rgba(255,255,255,0.5)"
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#0050f0",
            "borderColor": "#ffffff",
            "borderRadius": 4,
            "borderStyle": "solid",
            "borderWidth": 1,
            "color": "white",
            "display": "flex",
            "fontFamily": "Roboto",
            "fontSize": 16,
            "fontStyle": "normal",
            "fontWeight": "normal",
            "height": 48,
            "marginTop": 16,
            "textAlign": "center",
            "width": 312,
          },
        ]
      }
      value=""
    />
    <TextInput
      onChangeText={[Function]}
      placeholder="Digite sua senha de acesso"
      placeholderTextColor="rgba(255,255,255,0.5)"
      secureTextEntry={true}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#0050f0",
            "borderColor": "#ffffff",
            "borderRadius": 4,
            "borderStyle": "solid",
            "borderWidth": 1,
            "color": "white",
            "display": "flex",
            "fontFamily": "Roboto",
            "fontSize": 16,
            "fontStyle": "normal",
            "fontWeight": "normal",
            "height": 48,
            "marginTop": 16,
            "textAlign": "center",
            "width": 312,
          },
        ]
      }
      value=""
    />
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFF",
        "borderRadius": 4,
        "height": 48,
        "justifyContent": "center",
        "left": 24,
        "opacity": 1,
        "top": 96,
        "width": 312,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "alignItems": "center",
            "color": "#0050F0",
            "display": "flex",
            "fontFamily": "Roboto",
            "fontSize": 16,
            "fontStyle": "normal",
            "fontWeight": "500",
            "textAlign": "center",
          },
        ]
      }
    >
      Entrar
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "height": 24,
          "justifyContent": "center",
          "top": 110,
          "width": "100%",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "rgba(255,255,255,0.5)",
            "fontFamily": "Red Hat Display",
            "fontSize": 14,
            "fontStyle": "normal",
            "fontWeight": "normal",
            "textAlign": "center",
          },
        ]
      }
    >
      NÃ£o possui um acesso?
    </Text>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "marginLeft": 5,
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FFF",
              "fontFamily": "Red Hat Display",
              "fontSize": 14,
              "fontStyle": "normal",
              "fontWeight": "normal",
              "textAlign": "center",
            },
          ]
        }
      >
        Cadastre-se aqui
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
